### ISSUE TITLE
Keycloak V25 breaks CAS client-scope creation

### ISSUE THREAD

**Issue by ValentinLorand** [jacekkow/keycloak-protocol-cas#119]
Helloo,

I was very happy to be able to create CAS client scope in keycloak 24.0.4 without any problem but since I upgraded Keycloak to version 25.0.4, I can't create client-scope for protocol CAS.

The CAS choice is present in the creation form but when I submit the form I get a 400 Bad Request response because of the bad protocol value (see the screenshot below) : 

![image](https://github.com/user-attachments/assets/eb6bad7c-28eb-4c46-b6d2-e8aa15d7e024)

The problem occurs both from the API and the admin console interface.

Do you have any idea of the origin of this problem and a potential fix in mind ? :) 

Best regards,
Valentin Lorand

**Comment by ValentinLorand** [comment]
After some digging, I found the origin of the problem :

![image](https://github.com/user-attachments/assets/ce476e55-3fb1-45a4-9096-896a93707113)

A fragment of code has been implemented in the version 25 of Keycloak where an hardcoded comparison is performed to validated the protocol  (SAML or OIDC). This implementation BREAKS the extensibility of Keycloak regarding the protocols.

cf : https://github.com/keycloak/keycloak/pull/29544

**Comment by jacekkow** [comment]
@ValentinLorand - would you report it upstream?

**Comment by ValentinLorand** [comment]
Yes, the issue is created here : https://github.com/keycloak/keycloak/issues/32281

**Comment by tsaarni** [comment]
xref https://github.com/keycloak/keycloak/pull/40251
